cohortextractor/tpp_backend.py

```diff
@@ -344,6 +344,16 @@ def get_case_expression(self, other_columns, column_type, category_definitions):
             raise ValueError(
                 "At least one category must be given the definition 'DEFAULT'"
             )
+
+        # If categories are date-like strings we don't want to apply the usual
+        # reformatting that we apply to dates to make sure that MSSQL
+        # interprets them consistently; instead we want them in the format we
+        # expect for outputs, which is their original ISO format. If these
```

Comment:
Found this comment hard to understand, initially. Something like this perhaps? "We pass the `reformat_dates=False` option to the quoting function here to preserve date-like strings in the user-supplied ISO format, as opposed to converting to MSSQL-friendly format. If these...."

Reply:
Yeah, it's a fundamentally nasty thing to try to explain. Happy to amend to whatever made it clearer for you.